akka-cluster-sharding-typed/src/main/scala/akka/cluster/sharding/typed/internal/ClusterShardingImpl.scala

```diff
@@ -377,6 +385,17 @@ import akka.util.JavaDurationConverters._
         message: T,
         @unused timeout: Timeout): Future[U] = {
       shardRegion ! ShardingEnvelope(entityId, message)
+
+      future.onComplete {
+        case Failure(ex) =>
+          logger.error(
```

Comment:
I don't think this is "the Akka way". We don't log ask timeouts (or Streams failures), that's up to the caller.
Isn't there already contextual information in the exception? See how the `refPrefix` is passed in to the EntityRefImpl.

Reply:
You are right. `refPrefix` is giving the context we need. I didn't notice it. 

I will also change in Lagom. Instead of always logging, I will change the exception message (if it's a AskTimeout) to include the missing context. As such the experience will be similar to Akka.